File: snippets/csharp/System.Numerics/BigInteger/LogMethod/log1.cs
﻿// <Snippet1>
using System;
using System.Numerics;

public class Example
{
   public static void Main()
   {
      BigInteger[] values = { 2, 100, BigInteger.Pow(1000, 100), 
                              BigInteger.Pow(2, 64) };
      foreach (var value in values)                                    
         Console.WriteLine("The square root of {0} is {1}", value, 
                           Math.Exp(BigInteger.Log(value) / 2));
   }
}
// The example displays the following output:
//    The square root of 2 is 1.41421356237309
//    The square root of 100 is 10
//    The square root of 1000000000000000000000000000000000000000000000000000000000000
//    00000000000000000000000000000000000000000000000000000000000000000000000000000000
//    00000000000000000000000000000000000000000000000000000000000000000000000000000000
//    00000000000000000000000000000000000000000000000000000000000000000000000000000000
//     is 9.99999999999988E+149
//    The square root of 18446744073709551616 is 4294967296
// </Snippet1>

File: snippets/fsharp/System.Numerics/Complex/Exp/log1.fs
// <Snippet1>
open System
open System.Numerics

let values =
    [ Complex(1.53, 9.26)
      Complex(2.53, -8.12)
      Complex(-2.81, 5.32)
      Complex(-1.09, -3.43)
      Complex(Double.MinValue / 2.0, Double.MinValue / 2.0) ]

for value in values do
    printfn $"Exp(Log({value}) = {Complex.Exp(Complex.Log(value))}"
// The example displays the following output:
//       Exp(Log((1.53, 9.26)) = (1.53, 9.26)
//       Exp(Log((2.53, -8.12)) = (2.53, -8.12)
//       Exp(Log((-2.81, 5.32)) = (-2.81, 5.32)
//       Exp(Log((-1.09, -3.43)) = (-1.09, -3.43)
//       Exp(Log((-8.98846567431158E+307, -8.98846567431158E+307)) = (-8.98846567431161E+307, -8.98846567431161E+307)
// </Snippet1>
